wp/headless-wp/includes/classes/Search/PostSearchHandler.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Search Class
+ *
+ * @package HeadlessWP
+ */
+
+namespace HeadlessWP\Search;
+
+/**
+ * Class PostSearchHandler
+ */
+class PostSearchHandler extends \WP_REST_Post_Search_Handler {
+
+	/**
+	 * Overrides the prepare_item_links method for the search result to include author and taxonomies embedded data.
+	 *
+	 * @param int $id Item ID.
+	 * @return array Links for the given item.
+	 */
+	public function prepare_item_links( $id ) {
+
+		$post  = get_post( $id );
```

Comment:
The item might not always be a post. It's possible to search for terms: https://js1.10up.com/wp-json/wp/v2/search?type=term

It would be better to check if it's actually a post first.

Reply:
added check to make sure it's a post